Sources/Foundation/ProcessInfo.swift

```diff
@@ -80,11 +80,68 @@ open class ProcessInfo: NSObject {
         return CFUUIDCreateString(kCFAllocatorSystemDefault, uuid)._swiftObject
     }
 
+#if os(Windows)
+    internal var _rawOperatingSystemVersionInfo: RTL_OSVERSIONINFOEXW? {
+        guard let ntdll = ("ntdll.dll".withCString(encodedAs: UTF16.self) {
+            LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
+        }) else {
+            return nil
+        }
+        defer { FreeLibrary(ntdll) }
+        typealias RTLGetVersionTy = @convention(c) (UnsafeMutablePointer<RTL_OSVERSIONINFOEXW>) -> NTSTATUS
+        guard let pfnRTLGetVersion = unsafeBitCast(GetProcAddress(ntdll, "RtlGetVersion"), to: Optional<RTLGetVersionTy>.self) else {
+            return nil
+        }
+        var osVersionInfo = RTL_OSVERSIONINFOEXW()
+        osVersionInfo.dwOSVersionInfoSize = DWORD(MemoryLayout<RTL_OSVERSIONINFOEXW>.size)
+        guard pfnRTLGetVersion(&osVersionInfo) == 0 else {
+            return nil
+        }
+        return osVersionInfo
+    }
+#endif
+
     open var operatingSystemVersionString: String {
         let fallback = "Unknown"
 #if os(Linux)
-        let version = try? String(contentsOf: URL(fileURLWithPath: "/proc/version_signature", isDirectory: false), encoding: .utf8)
-        return version ?? fallback
+        var utsNameBuffer = utsname()
+        guard uname(&utsNameBuffer) == 0 else {
+            return fallback
+        }
+        let release = withUnsafePointer(to: &utsNameBuffer.release.0) { String(cString: $0) }
+
+        return release
```

Comment:
Might be worth adding a check that `guard release != "" else { return fallback }` just incase `utsname()` breaks.